src/Raven.Server/Documents/Handlers/Processors/Documents/AbstractDocumentHandlerProcessorForGet.cs

```diff
@@ -0,0 +1,451 @@
+            {
+                writer.WriteComma();
+                writer.WritePropertyName(nameof(GetDocumentsResult.CounterIncludes));
+
+                await writer.WriteCountersAsync(result.CounterIncludes, CancellationToken);
+            }
+
+            if (result.TimeSeriesIncludes?.Count > 0)
+            {
+                writer.WriteComma();
+                writer.WritePropertyName(nameof(GetDocumentsResult.TimeSeriesIncludes));
+                await writer.WriteTimeSeriesAsync(result.TimeSeriesIncludes, CancellationToken);
+            }
+
+            if (result.RevisionsChangeVectorIncludes?.Count > 0 || result.IdByRevisionsByDateTimeIncludes?.Count > 0)
+            {
+                writer.WriteComma();
+                writer.WritePropertyName(nameof(GetDocumentsResult.RevisionIncludes));
+                writer.WriteStartArray();
+                await writer.WriteRevisionIncludes(context: context, revisionsByChangeVector: result.RevisionsChangeVectorIncludes, revisionsByDateTime: result.IdByRevisionsByDateTimeIncludes, CancellationToken);
+                writer.WriteEndArray();
+            }
+
+            if (result.CompareExchangeIncludes?.Count > 0)
+            {
+                writer.WriteComma();
+                writer.WritePropertyName(nameof(GetDocumentsResult.CompareExchangeValueIncludes));
+                await writer.WriteCompareExchangeValuesAsync(result.CompareExchangeIncludes, CancellationToken);
+            }
+
+            writer.WriteEndObject();
+        }
+        return (numberOfResults, totalDocumentsSizeInBytes);
+    }
+
+    protected abstract ValueTask<(long NumberOfResults, long TotalDocumentsSizeInBytes)> WriteDocumentsAsync(AsyncBlittableJsonTextWriter writer, TOperationContext context, string propertyName,
+        List<TDocumentType> documentsToWrite, bool metadataOnly, CancellationToken token);
+
+    protected abstract ValueTask WriteIncludesAsync(AsyncBlittableJsonTextWriter writer, TOperationContext context, string propertyName,
+        List<TDocumentType> includes, CancellationToken token);
+
+    protected abstract ValueTask<DocumentsByIdResult<TDocumentType>> GetDocumentsByIdImplAsync(TOperationContext context, StringValues ids,
+        StringValues includePaths, RevisionIncludeField revisions, StringValues counters, HashSet<AbstractTimeSeriesRange> timeSeries, StringValues compareExchangeValues, bool metadataOnly, string etag);
+
+    protected abstract ValueTask HandleMissingIncludes(TOperationContext context, DocumentsByIdResult<TDocumentType> result, bool metadataOnly);
+
+    protected async ValueTask<(long NumberOfResults, long TotalDocumentsSizeInBytes)> GetDocumentsAsync(TOperationContext context, long? etag, StartsWithParams startsWith, bool metadataOnly, string changeVector)
+    {
+        var result = await GetDocumentsImplAsync(context, etag, startsWith, changeVector);
+
+        if (changeVector == result.Etag)
```

Comment:
is this valid for the sharded version as well?

Reply:
Yes